android-sdk-simplexml/src/main/java/com/microsoft/hsg/android/simplexml/client/HealthVaultRestClient.java

```diff
@@ -0,0 +1,163 @@
+import org.joda.time.LocalDateTime;
+import org.joda.time.Minutes;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserFactory;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.util.Base64;
+
+import com.microsoft.hsg.Authenticator;
+import com.microsoft.hsg.Connection;
+import com.microsoft.hsg.HVException;
+import com.microsoft.hsg.Transport;
+import com.microsoft.hsg.URLConnectionTransport;
+import com.microsoft.hsg.android.simplexml.HealthVaultSettings;
+import com.microsoft.hsg.android.simplexml.HealthVaultFileSettings;
+import com.google.common.reflect.TypeToken;
+import com.microsoft.hsg.android.simplexml.things.types.types.Record;
+import com.microsoft.rest.RestClient;
+import com.microsoft.rest.ServiceCallback;
+import com.microsoft.rest.ServiceClient;
+import com.microsoft.rest.ServiceFuture;
+import com.microsoft.rest.ServiceResponse;
+import com.microsoft.rest.Validator;
+
+import healthvault.client.implementation.MicrosoftHealthVaultRestApiImpl;
+
+import healthvault.client.models.ActionPlan;
+import healthvault.client.models.ActionPlansResponseActionPlanInstance;
+import healthvault.client.models.Objective;
+import okhttp3.ConnectionPool;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
+import retrofit2.converter.gson.GsonConverterFactory;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+import retrofit2.http.Body;
+import retrofit2.http.GET;
+import retrofit2.http.HTTP;
+import retrofit2.http.Headers;
+import retrofit2.http.PATCH;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+
+public class HealthVaultRestClient implements IHealthVaultRestClient {
+
+	private static MicrosoftHealthVaultRestApiImpl instance;
+	private static DateTime mLastRefreshedSessionCredential = DateTime.now();;
+	private static final int mSessionCredentialCallThresholdMinutes = 5;
+	private static Retrofit mRetrofit;
+	private static String mRestURL;
+	private static OkHttpClient.Builder mOkBuilder;
+	private static Connection mConnection;
+	private static Record mCurrentRecord;
+	private static HealthVaultSettings mSettings;
+
+	public HealthVaultRestClient (HealthVaultSettings settings, Connection connection, Record currentRecord){
+		Initialize(settings, connection, currentRecord);
+	}
+
+	private void Initialize(HealthVaultSettings settings, Connection connection, Record currentRecord){
+		if (connection == null){
+			throw new HVException("connection is null");
+		}
+		mSettings = settings;
+		mRestURL = settings.getRestUrl();
+		mCurrentRecord = currentRecord;
+		mConnection = connection;
+		mOkBuilder = getOkHttp(connection, currentRecord);
+		mRetrofit = getRetrofit(mRestURL);
+		tokenRefresh(connection);
```

Comment:
>tokenRefresh [](start = 2, length = 12)

you are doing this as part of ctor .. making call to network .. wondering y don't getOkHttp instead call in the token refresh ..